### PR TITLE
Feature: reduce webpack config

### PR DESCRIPTION
**Pull request by rjanjua**
The idea behind this PR is to minimise the amount of config we required to get web running.

The main files to pay attention to would be the config files in .fructose.

Whereas before we had a whole load of config in `.fructose/web` that wasn't directly relevant to the project under test, we now only need to include a small amount of config that is directly related to the project we are testing. In addition to that we can add a head.html file to include any additional html like styles and fonts in the app.

The files we can add are:
  - webpack.config.js 
  - vendor.webpack.config.js
  - head.html
  - index.web.js

You can compare the [before](https://github.com/newsuk/fructose/tree/master/e2eTests/.fructose) and [after](https://github.com/newsuk/fructose/tree/feat/reduce-webpack-config/e2eTests/.fructose)

### CODE REVIEW

**Comment by rjanjua** [comment]
Tested this against times-components and it loads up the fructose app, and loads up components

File: packages/web/bin/start [hunk 1]
```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env node
```
Comment (jackharvey1) [inline packages/web/bin/start]: Can this file be `.js` so that eslint will be run on it?

File: e2eTests/.fructose/jest-setup.js [hunk 2]
```diff
@@ -5,20 +5,26 @@ import webdriverio from "webdriverio";
...
+const isWebStarted = port =>
   new Promise(resolve => {
-    portscanner.checkPortStatus(port, "127.0.0.1", (error, status) => {
-      // Status is 'open' if currently in use or 'closed' if available
-      const isTaken = status === "open";
-      resolve(isTaken);
+    request({
+      uri:`http://localhost:${port}`,
+      timeout: 200
+    }, function (error, response, body) {
```
Comment (jackharvey1) [inline e2eTests/.fructose/jest-setup.js]: Could we pull these out?

File: e2eTests/.fructose/vendor.webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+/* eslint-disable */
```
Comment (jackharvey1) [inline e2eTests/.fructose/vendor.webpack.config.js]: How come eslint was disabled?

File: e2eTests/.fructose/webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+module.exports = {
+  entry: {
+    app: path.join(__dirname, "./index.web.js")
+  },
+  resolve: {
+    alias: {
+      "react-native": "react-native-web"
+    },
+    extensions: [".web.js", ".js", ".ios.js", ".android.js"],
+    mainFields: ["module", "main"],
```
Comment (jackharvey1) [inline e2eTests/.fructose/webpack.config.js]: Unless this is a config I've never seen before, I think plugins are supposed to be on the same level as `resolve`, so there would be a `}` here.

Comment (rjanjua) [reply e2eTests/.fructose/webpack.config.js]: It is supposed to be like this.

Comment (jackharvey1) [reply e2eTests/.fructose/webpack.config.js]: How come it's different? Can't find anything on Google.

File: e2eTests/.fructose/jest-setup.js [hunk 5]
```diff
@@ -77,12 +44,7 @@ beforeAll(async () => {
       await detox.init(config.detox);
     }
   } else if (process.env.WEB) {
-    await fructose.hooks.web.setup();
-    const taken = await checkIfPortTaken(10);
-    if (!taken) {
-      console.error("The fructose app did not start correctly");
-      process.exit(1);
-    }
+    await fructose.hooks.web.setup(3000, 20000);
```
Comment (craigbilner) [inline e2eTests/.fructose/jest-setup.js]: optional: `{port: 3000, healthCheckTimeout: 20000}`

File: e2eTests/.gitignore [hunk 6]
```diff
@@ -55,6 +55,11 @@ fastlane/screenshots
 **/.fructose/components.js
 **/.fructose/components.test.js
 
+!dist
```
Comment (craigbilner) [inline e2eTests/.gitignore]: the whole of dist should probably be ignored and you can check in assets separately

File: packages/web/bin/start.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+if (!program.buildDir)
+  throw new Error(
+    "you must define the build directory: --build-dir [directory]"
+  );
+
+const merge = require("webpack-merge");
+const webpack = require("webpack");
+const WebpackDevServer = require("webpack-dev-server");
+const path = require("path");
+const upperConfig = require("../../../../../../.fructose/webpack.config.js"); // eslint-disable-line import/no-unresolved
```
Comment (craigbilner) [inline packages/web/bin/start.js]: all the nesting?

**Pull request merged**